### ISSUE TITLE
Build and keep a cache of installed modules for intellisense

### ISSUE THREAD

**Issue by gerane** [PowerShell/PowerShellEditorServices#263]
It would be useful if PSES would automatically cache any modules in the PSModulePath so that it could offer intellisense without needing to load the module.

**Comment by daviwil** [comment]
Yeah, I'd have to find a way to do that securely, i.e. without loading the modules (and causing the .psm1 to be loaded).

**Comment by gerane** [comment]
Couldn't you use something like
``` [System.Management.Automation.PSParser]::Tokenize()
```

**Comment by daviwil** [comment]
That'd mean I'd have to manually scan every module in the module path and parse each code file contained within.  Not sure if that's an ideal way to do it.  I'll ask someone in the team to see if there's a better way

**Comment by kilasuit** [comment]
@daviwil - Perhaps you can make use the current cache as detailed at the bottom of https://msdn.microsoft.com/en-us/powershell/wmf/5.1/engine-improvements

**Comment by andyleejordan** [comment]
Not planned — a separate module cache is incompatible with the current `TabExpansion2`-based completion architecture. Closing as obsolete.

---
_Drafted by GitHub Copilot (Claude Opus 4.8) at @andyleejordan's request._
